cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -552,6 +559,7 @@ public void executeReportGeneration(HttpServiceRequest request, HttpServiceRespo
         reportRequest = decodeRequestBody(requestJson, REPORT_GENERATION_REQUEST_TYPE);
         reportRequest.validate();
       } catch (IllegalArgumentException e) {
+        LOG.error("Invalid report generation request {}.", requestJson, e);
```

Comment:
Shouldn't log an error on user request error. Otherwise it opens up a hole for user to flood the log.